lib/parser/ruby33.y

```diff
@@ -0,0 +1,3238 @@
+                |
+                    {
+                      result = []
+                    }
+
+       singleton: var_ref
+                | tLPAREN2 expr rparen
+                    {
+                      result = val[1]
+                    }
+
+      assoc_list: # nothing
+                    {
+                      result = []
+                    }
+                | assocs trailer
+
+          assocs: assoc
+                    {
+                      result = [ val[0] ]
+                    }
+                | assocs tCOMMA assoc
+                    {
+                      result = val[0] << val[2]
+                    }
+
+           assoc: arg_value tASSOC arg_value
+                    {
+                      result = @builder.pair(val[0], val[1], val[2])
+                    }
+                | tLABEL arg_value
+                    {
+                      result = @builder.pair_keyword(val[0], val[1])
+                    }
+                | tLABEL
+                    {
+                      result = @builder.pair_label(val[0])
+                    }
+                | tSTRING_BEG string_contents tLABEL_END arg_value
+                    {
+                      result = @builder.pair_quoted(val[0], val[1], val[2], val[3])
+                    }
+                | tDSTAR arg_value
+                    {
+                      result = @builder.kwsplat(val[0], val[1])
+                    }
+                | tDSTAR
+                    {
+                      if !@static_env.declared_anonymous_kwrestarg?
+                        diagnostic :error, :no_anonymous_kwrestarg, nil, val[0]
+                      end
+
+                      result = @builder.forwarded_kwrestarg(val[0])
+                    }
+
+       operation: tIDENTIFIER | tCONSTANT | tFID
+      operation2: operation | op
+      operation3: tIDENTIFIER | tFID | op
+    dot_or_colon: call_op | tCOLON2
+         call_op: tDOT
+                    {
+                      result = [:dot, val[0][1]]
+                    }
+                | tANDDOT
+                    {
+                      result = [:anddot, val[0][1]]
+                    }
+       opt_terms:  | terms
+          opt_nl:  | tNL
+          rparen: opt_nl tRPAREN
+                    {
+                      result = val[1]
+                    }
+        rbracket: opt_nl tRBRACK
+                    {
+                      result = val[1]
+                    }
+          rbrace: opt_nl tRCURLY
+                    {
+                      result = val[1]
+                    }
+         trailer: opt_nl | tCOMMA
+
+            term: tSEMI
+                  {
+                    yyerrok
+                  }
+                | tNL
+
+           terms: term
+                | terms tSEMI
+
+            none: # nothing
+                  {
+                    result = nil
+                  }
+end
+
+---- header
+
+require 'parser'
+
+---- inner
+
+  def version
+    32
```

Comment:
This should be 33

Reply:
Ouch, I didn't notice that one. Thanks! And fixed.